include/bluetooth/mesh/main.h

```diff
@@ -230,6 +272,62 @@ int bt_mesh_input_string(const char *str);
  */
 int bt_mesh_input_number(uint32_t num);
 
+/** @brief Provide Device public key.
+ *
+ *  @param public_key Device public key.
+ *
+ *  @return Zero on success or (negative) error code otherwise.
+ */
+int bt_mesh_prov_remote_pub_key_set(const uint8_t public_key[64]);
+
+/** @brief Input OOB authentication is used.
+ *
+ *  This is intend to set input oob authentication method which be use after
+ *  @ref bt_mesh_prov::capabilities has been called.
+ *
+ *  @param action Authentication action will be used by device.
+ *  @param size Authentication size will be used by device and provisioner.
+ *
+ *  @return Zero on success or (negative) error code otherwise.
+ */
+int bt_mesh_auth_method_set_input(bt_mesh_input_action_t action, uint8_t size);
+
+/** @brief Output OOB authentication is used.
+ *
+ *  This is intend to set ouput oob authentication method which be use after
+ *  @ref bt_mesh_prov::capabilities has been called.
+ *
+ *  @param action Authentication action will be used by device.
+ *  @param size Authentication size will be used by device and provisioner.
+ *
+ *  @return Zero on success or (negative) error code otherwise.
+ */
+int bt_mesh_auth_method_set_output(bt_mesh_output_action_t action, uint8_t size);
+
+/** @brief Static OOB authentication is used.
+ *
+ *  This is intend to set static oob authentication method which be use after
+ *  @ref bt_mesh_prov::capabilities has been called.
+ *
+ *  @param static_val Static OOB value.
+ *  @param size Authentication size will be used by device and provisioner.
+ *
+ *  @return Zero on success or (negative) error code otherwise.
+ */
+int bt_mesh_auth_method_set_static(const uint8_t *static_val, uint8_t size);
+
+/** @brief No OOB authentication is used.
+ *
+ *  This is intend to set no oob authentication method which be use after
+ *  @ref bt_mesh_prov::capabilities has been called.
+ *
+ *  When @ref bt_mesh_prov::capabilities set to NULL or not set any method
+ *  before, this method will be used.
+ *
+ *  @return Zero on success or (negative) error code otherwise.
+ */
+int bt_mesh_auth_method_set_none(void);
```

Comment:
```suggestion
/** @brief Don't use OOB authentication.
 *
 *  Provisioner only.
 *  
 *  Don't use any authentication when provisioning new devices. This is the 
 *  default behavior.
 *  
 *  @warning Not using any authentication exposes the mesh network to 
 *           impersonation attacks, where attackers can pretend to be the 
 *           unprovisioned device to gain access to the network. Authentication
 *           is strongly encouraged.
 *           
 *  @return Zero on success or (negative) error code otherwise.
 */
int bt_mesh_auth_method_set_none(void);
```